test/client/fixtures/sandbox/xhr-test.js

```diff
@@ -201,3 +205,26 @@ asyncTest('authorization headers by client should be processed (GH-1016)', funct
     });
     xhr.send();
 });
+
+asyncTest('our internal the onreadystatechange handler must be first (GH-1283)', function () {
+    var xhr      = new XMLHttpRequest();
+    var timeout  = null;
+    var testDone = function (eventObj) {
+        clearTimeout(timeout);
+        ok(!!eventObj);
+        start();
+    };
+
+    var readyStateChangeHandler = function (e) {
+        if (this.readyState === this.DONE)
+            e.stopImmediatePropagation();
```

Comment:
add `ok(false, 'outer handler is called before XHR_COMPLETED_EVENT');`

Reply:
this handler will be called in all cases

Reply:
Ok